common/txmgr/inmemory_store.go

```diff
@@ -79,11 +81,20 @@ func NewInMemoryStore[
 		ms.maxUnstarted = 10000
 	}
 
+	addressesToTxs := map[ADDR][]txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}
+	// populate all enabled addresses
+	enabledAddresses, err := keyStore.EnabledAddressesForChain(ctx, chainID)
```

Comment:
I left a similar comment on one of the other PRs but is this necessary if we load using from address from all tx below?